packages/database/src/modelClasses/Entity.js

```diff
@@ -374,6 +374,38 @@ export class EntityModel extends DatabaseModel {
     );
   }
 
+  async getEntityAnswerWithEntityName(dataSourceEntities, questionCode) {
+    const eventId = 'event_id';
+    const entityAnswerIdToName = 'entity_answer_id_to_name';
+    const entityAnswerIdToNameMapping = await this.database.executeSql(
+      `
+        SELECT sr.id as ${eventId}, e2.name as ${entityAnswerIdToName}
```

Comment:
There are some valid reasons for using raw queries: performance optimisations, specific operations that our `DatabaseModel` API does not support etc.

None of those reasons apply here, so we should use the `models` object which is passed around as an abstraction for Model operations - you'll find a lot of example for this in our codebase

Reply:
Thanks for sharing the specific purposes of this class.